arcgis-ios-sdk-samples/Layers/Group layers/LayersTableViewController.swift

```diff
@@ -16,42 +16,86 @@ import UIKit
 import ArcGIS
 
 class LayersTableViewController: UITableViewController, GroupLayersCellDelegate, GroupLayersSectionViewDelegate {
-    var layers = [AGSLayer]()
+    var layers = [AGSGroupLayer]()
+    var tableViewContentSizeObservation: NSKeyValueObservation?
     
     override func viewDidLoad() {
         super.viewDidLoad()
         self.tableView.register(GroupLayersSectionView.nib, forHeaderFooterViewReuseIdentifier: GroupLayersSectionView.reuseIdentifier)
     }
     
+    // Adjust the size of the table view according to its contents.
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        tableViewContentSizeObservation = tableView.observe(\.contentSize) { [unowned self] (tableView, _) in
+            self.preferredContentSize = CGSize(width: self.preferredContentSize.width, height: tableView.contentSize.height)
+        }
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        tableViewContentSizeObservation = nil
+    }
+    
     // MARK: - UITableViewDataSource
     
     override func numberOfSections(in tableView: UITableView) -> Int {
         return layers.count
     }
     
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        guard let groupLayer = layers[section] as? AGSGroupLayer else { return 0 }
+        let groupLayer = layers[section]
         return groupLayer.layers.count
     }
     
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath) as! GroupLayersCell
-        
-        guard let groupLayer = layers[indexPath.section] as? AGSGroupLayer, let childLayers = groupLayer.layers as? [AGSLayer] else { return cell }
-        
-        let childLayer = childLayers[indexPath.row]
-        
-        // Set label.
-        cell.layerNameLabel.text = formattedValue(of: childLayer.name)
-        
-        // Set state of the switch.
-        cell.layerVisibilitySwitch.isOn = childLayer.isVisible
-        cell.layerVisibilitySwitch.isEnabled = groupLayer.isVisible
-        
-        // To update the visibility of operational layers on switch toggle.
-        cell.delegate = self
-        
-        return cell
+        let groupLayer = layers[indexPath.section]
+        switch groupLayer.visibilityMode {
+        case .independent:
+            let cell = tableView.dequeueReusableCell(withIdentifier: "switchCell", for: indexPath) as! GroupLayersCell
+            let childLayers = groupLayer.layers as! [AGSLayer]
+            let childLayer = childLayers[indexPath.row]
```

Comment:
It also looks like this can be moved outside of the switch statement since it is the same for all cases.